src/components/Nav/Nav.Props.ts

```diff
@@ -78,17 +78,21 @@ export interface INavLink {
   isExpanded?: boolean;
 
   /**
-   * Any additional properties to apply to the rendered links.
+   * (Optional) Aria label for nav link
    */
-  [propertyName: string]: any;
+  ariaLabel?: string;
 
   /**
-   * (Optional) Aria label for nav link
+   * (Optional) Unique ID for nav link
    */
-  ariaLabel?: string;
+  id?: string;
```

Comment:
I am super sorry for suggesting "id" but I think you should rename this back to "key". I didn't realize you were talking about INavLink, thought you mentioned a component's props interface. "key" should not be used in props interfaces, but "key" can and probably should be used in object interfaces. It describes the unique value for the object that should be used when rendering the ui representation of the object.